results.nim

```diff
@@ -303,34 +303,40 @@ type
     ## https://github.com/nim-lang/Nim/issues/14318 - generic error raises pragma
 
     # TODO https://github.com/nim-lang/Nim/issues/20699
-    # case o: bool
+    # case oResultPrivate: bool
     # of false:
-    #   e: E
+    #   eResultPrivate: E
     # of true:
-    #   v: T
-
+    #   vResultPrivate: T
+
+    # TODO ResultPrivate works around
+    # * https://github.com/nim-lang/Nim/issues/3770
+    # * https://github.com/nim-lang/Nim/issues/20900
```

Comment:
fyi, there are PRs for these two issues:

https://github.com/nim-lang/Nim/pull/21554
https://github.com/nim-lang/Nim/pull/21649

Reply:
nice! hopefully we can get a backport and release cc @narimiran